### PR TITLE
Refactor JaxToolboxSlurmCommandGenStrategy

### PR DESCRIPTION
**Pull request by TaekyungHeo**
## Summary
Refactor container load and run command generation into separate functions

## Test Plan
1. CI passes
2. Manual run

### CODE REVIEW

**Comment by TaekyungHeo** [comment]
Yes it was what I used, @srivatsankrishnan

File: src/cloudai/schema/test_template/jax_toolbox/slurm_command_gen_strategy.py [hunk 1]
```diff
@@ -423,32 +399,45 @@ def _generate_pre_test_check_command(self, cmd_args: Dict[str, str], output_path
...
+    def _generate_container_load_command(self, slurm_args: Dict[str, Any]) -> List[str]:
+        """Generate the command for loading a container."""
+        commands = []
+        container_image = slurm_args.get("image_path")
+        if not container_image:
+            raise ValueError("image_path in slurm_args must be a valid path")
 
-        Args:
-            container_image (str): The path to the container image.
+        commands.append('if [ "$PRE_TEST_SUCCESS" = true ]; then')
```
Comment (amaslenn) [inline src/cloudai/schema/test_template/jax_toolbox/slurm_command_gen_strategy.py:419]: Let's avoid doing multiple function call and use list creation as `l = [...]`, same for the code below.

Comment (amaslenn) [reply src/cloudai/schema/test_template/jax_toolbox/slurm_command_gen_strategy.py]: Sorry, I wasn't clear. Let's change 
```py
    ...
    commands.append('    echo "Running srun command"')
    commands.append("    srun \\")
    commands.append("    --mpi=none \\")
    ...
```
to
```py
cmds = [
    ...
    '    echo "Running srun command"',
    "    srun \\",
    "    --mpi=none \\",
    ...
]
```

Comment (TaekyungHeo) [reply src/cloudai/schema/test_template/jax_toolbox/slurm_command_gen_strategy.py]: Thanks, updated.

**Pull request merged**